XenAdmin/Wizards/PatchingWizard/AutomatedUpdatesBasePage.cs

```diff
@@ -512,8 +506,12 @@ private void RetryFailedActions()
 
         private void SkipFailedActions()
         {
-            var msg = string.Join(Environment.NewLine,
-                failSkippableWorkers.Select(w => ((PlanActionWithSession)w.PlanActionMarkedToSkip).SkipMessage));
+            var skippableWorkers = failedWorkers.Where(w => w.FirstFailedSkippableAction != null);
+            foreach (var w in skippableWorkers)
+            {
+                System.Diagnostics.Trace.Assert(!string.IsNullOrEmpty(w.FirstFailedSkippableAction.Title));

```

Comment:
Let's move this to where we set FirstFailedSkippableAction, and change it to Debug.Assert